File: back-end/spms/src/main/java/com/revature/spms/entity/StoryPitch.java
package com.revature.spms.entity;

import lombok.Data;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Table(name = "story_pitch")
@Data
public class StoryPitch {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "author_id")
    private Long authorId;

    @Column(name = "title")
    private String title;

    @Column(name = "genre")
    private String genre;

    @Column(name = "tag")
    private String tag;

    @Column(name = "description")
    private String description;

    @Column(name = "date_created")
    private Date dateCreated;

    @Column(name = "last_updated")
    private Date lastUpdated;

    @Column(name = "completion_date")
    private Date completionDate;

    @Column(name = "word_count")
    private int wordCount;

    @Column(name="points")
    private int points;

    @Column(name = "approved")
    private char approved;

    @Column(name = "on_hold")
    private char onHold;

    @Column(name = "completed")
    private char completed;
}

File: front-end/spms-angular/src/app/models/story-pitch.ts
import { Genre } from "./genre";
import { StoryType } from "./story-type";

export class StoryPitch {
   constructor(
    public id: number,
    public authorId: number,
    public title: string,
    public genre: Genre,
    public tag: string,
    public description: string,
    public points: number,
    public dateCreated: Date,
    public lastUpdated: Date,
    public completionDate: Date,
    public approved: string,
    public onHold: string,
    public completed: string,
    public storyType: StoryType, 
   ) {} 
    
    // constructor(    
    //     public id: number,    
    //     public authorId: number,
    //     public title: string,    
    //     public completionDate: Date,    
    //     public wordCount: number,
    //     public completed: string,
    //     public genre: string,    
    //     public tag: string,
    //     public description: string,
    //     public approved: string,    
    //     public on_hold: string
    //   ) {}

}
